src/providers/profile/profile.ts

```diff
@@ -926,6 +926,14 @@ export class ProfileProvider {
     this.logger.info('Importing Wallet xPrivKey');
     opts.xPrivKey = xPrivKey;
     return this.serverAssistedImport(opts).then(data => {
+      const key = this.keyProvider.getKeyByFingerprint(data.key);
+      if (key) {
```

Comment:
a comment would help help. // If we have the key already, bind the new wallets to the current key.